crypto/rand/rand_vms.c

```diff
@@ -40,6 +40,32 @@
 #  error "Unsupported seeding method configured; must be os"
 # endif
 
+#define PUBLIC_VECTORS "SYS$LIBRARY:SYS$PUBLIC_VECTORS.EXE"
+
+/*
+ * Helper functions
+ * ----------------
+ *
+ * gettime() gets time using sys$gettim_prec or if that's not available,
+ * sys$gettim.  sys$gettim_prec exists since OpenVMS v8.4.
+ */
+#define GETTIM_PREC "SYS$GETTIM_PREC"
+
+static int gettime(unsigned __int64 *t)
+{
+    static int gettim_prec_address_flag = 0;
+    static int (*gettim_prec_address)(unsigned __int64 *t) = NULL;
+
+    if (!gettim_prec_address_flag) {
```

Comment:
Why not:
```c
if (gettim_prec_address == NULL)
```
here and saving a flag/status variable?

Reply:
So that, if SYS$GETTIM_PREC isn't not available, gettime() will try to find it again every time it's called?